lib/compat/wordpress-6.6/block-template-utils.php

```diff
@@ -360,3 +360,111 @@ function gutenberg_get_block_templates( $query = array(), $template_type = 'wp_t
 	 */
 	return apply_filters( 'get_block_templates', $query_result, $query, $template_type );
 }
+
+/**
+ * Creates an export of the current templates and
+ * template parts from the site editor at the
+ * specified path in a ZIP file.
+ *
+ * @since 5.9.0
+ * @since 6.0.0 Adds the whole theme to the export archive.
+ *
+ * @global string $wp_version The WordPress version string.
+ *
+ * @return WP_Error|string Path of the ZIP file or error on failure.
+ */
+function gutenberg_generate_block_templates_export_file() {
```

Comment:
This is all just copy paste and swapping out the WP functions for the Gutenberg juicy goodness.